### PR TITLE
test: Code Coverage: Add Edit Expense Page #3

### PR DESCRIPTION
**Pull request by jayfyle**
### <samp>🤖 Generated by Copilot at 6568817</samp>

This pull request adds and updates mock data and unit tests for the add-edit-expense page. It also refactors the code in the `add-edit-expense.page.ts` file to extract some common logic into separate methods and files. The main purpose of these changes is to improve the testability and readability of the add-edit-expense feature.

### <samp>🤖 Generated by Copilot at 6568817</samp>

> _Mock data for tests_
> _`add-edit-expense` refactored_
> _Autumn of clean code_

### <samp>🤖 Generated by Copilot at 6568817</samp>

*  Add mock data for action sheet options, parsed receipt, and popover parameters in new files `action-sheet-options.data.ts`, `parsed-receipt.data.ts`, and `popover-params.data.ts` ([link](https://github.com/fylein/fyle-mobile-app/pull/2145/files?diff=unified&w=0#diff-a62585052482cdfd703e0c1a2418a8fbdc364a2330e303f93f4bd5722abeacc4R1-R22), [link](https://github.com/fylein/fyle-mobile-app/pull/2145/files?diff=unified&w=0#diff-c7abedba9661487444ab115563a479af552ae5d0c5aabfd882e1226a85c6cb94R1-R41), [link](https://github.com/fylein/fyle-mobile-app/pull/2145/files?diff=unified&w=0#diff-70865470c8ccd6234f34591bfd0679d67edaf6e5a6ac2e13708fb3eeb4e38b81R1-R13))
* Add import statements for existing mock data in `duplicate-sets.data.ts`, `org-settings.data.ts`, and `undo-merge.data.ts` ([link](https://github.com/fylein/fyle-mobile-app/pull/2145/files?diff=unified&w=0#diff-08994ab67f55c1a7ea9ee90f7380a90fbd85abd94db9dbeee14f217f2cb1ebdfR2), [link](https://github.com/fylein/fyle-mobile-app/pull/2145/files?diff=unified&w=0#diff-1cac010b51c1a7c2fbdf9bf224a7e267779365abe08b0f16df1ef46e41483951R3), [link](https://github.com/fylein/fyle-mobile-app/pull/2145/files?diff=unified&w=0#diff-c985d0e3a5de79886b9f77ef460522d8782df0d18dcd58777ee5805dbad0c20cR2))
* Add new mock data objects for org settings with corporate credit card disabled, unflattened transaction with all fields, and unflattened transaction with extracted data in `org-settings.data.ts`, `unflattened-expense.data.ts`, and `unflattened-txn.data.ts` ([link](https://github.com/fylein/fyle-mobile-app/pull/2145/files?diff=unified&w=0#diff-1cac010b51c1a7c2fbdf9bf224a7e267779365abe08b0f16df1ef46e41483951R1191-R1199), [link](https://github.com/fylein/fyle-mobile-app/pull/2145/files?diff=unified&w=0#diff-db737e50777751787c68eda6c5be391ed672eb00607a92de6aaf9c945a3010d6R226-R335), [link](https://github.com/fylein/fyle-mobile-app/pull/2145/files?diff=unified&w=0#diff-21ae928ce640ad62cfada5fee7bf0bbc490300ab04a25f8309a9878e21b2f1cfR302-R472))
* Refactor the code in `add-edit-expense.page.ts` to extract the logic for popover parameters, snackbar toast messages, form validation errors, and action sheet options into separate methods and move them to `action-sheet-options.data.ts` ([link](https://github.com/fylein/fyle-mobile-app/pull/2145/files?diff=unified&w=0#diff-492aac4c505a9ef8fb102b40f692554c3d0dc493b7abddc97b6665166efcc83bL644-R645), [link](https://github.com/fylein/fyle-mobile-app/pull/2145/files?diff=unified&w=0#diff-492aac4c505a9ef8fb102b40f692554c3d0dc493b7abddc97b6665166efcc83bL659-R667), [link](https://github.com/fylein/fyle-mobile-app/pull/2145/files?diff=unified&w=0#diff-492aac4c505a9ef8fb102b40f692554c3d0dc493b7abddc97b6665166efcc83bL688-R712), [link](https://github.com/fylein/fyle-mobile-app/pull/2145/files?diff=unified&w=0#diff-492aac4c505a9ef8fb102b40f692554c3d0dc493b7abddc97b6665166efcc83bL731-R751), [link](https://github.com/fylein/fyle-mobile-app/pull/2145/files?diff=unified&w=0#diff-492aac4c505a9ef8fb102b40f692554c3d0dc493b7abddc97b6665166efcc83bL742-R762), [link](https://github.com/fylein/fyle-mobile-app/pull/2145/files?diff=unified&w=0#diff-492aac4c505a9ef8fb102b40f692554c3d0dc493b7abddc97b6665166efcc83bR779-R814), [link](https://github.com/fylein/fyle-mobile-app/pull/2145/files?diff=unified&w=0#diff-492aac4c505a9ef8fb102b40f692554c3d0dc493b7abddc97b6665166efcc83bL797-R849), [link](https://github.com/fylein/fyle-mobile-app/pull/2145/files?diff=unified&w=0#diff-492aac4c505a9ef8fb102b40f692554c3d0dc493b7abddc97b6665166efcc83bL810-R856), [link](https://github.com/fylein/fyle-mobile-app/pull/2145/files?diff=unified&w=0#diff-492aac4c505a9ef8fb102b40f692554c3d0dc493b7abddc97b6665166efcc83bL823-R863), [link](https://github.com/fylein/fyle-mobile-app/pull/2145/files?diff=unified&w=0#diff-492aac4c505a9ef8fb102b40f692554c3d0dc493b7abddc97b6665166efcc83bL838-R872), [link](https://github.com/fylein/fyle-mobile-app/pull/2145/files?diff=unified&w=0#diff-492aac4c505a9ef8fb102b40f692554c3d0dc493b7abddc97b6665166efcc83bL847-R879), [link](https://github.com/fylein/fyle-mobile-app/pull/2145/files?diff=unified&w=0#diff-492aac4c505a9ef8fb102b40f692554c3d0dc493b7abddc97b6665166efcc83bL857-R887), [link](https://github.com/fylein/fyle-mobile-app/pull/2145/files?diff=unified&w=0#diff-492aac4c505a9ef8fb102b40f692554c3d0dc493b7abddc97b6665166efcc83bL889-L897), [link](https://github.com/fylein/fyle-mobile-app/pull/2145/files?diff=unified&w=0#diff-492aac4c505a9ef8fb102b40f692554c3d0dc493b7abddc97b6665166efcc83bL2950-R2972), [link](https://github.com/fylein/fyle-mobile-app/pull/2145/files?diff=unified&w=0#diff-492aac4c505a9ef8fb102b40f692554c3d0dc493b7abddc97b6665166efcc83bL2999-R3018), [link](https://github.com/fylein/fyle-mobile-app/pull/2145/files?diff=unified&w=0#diff-492aac4c505a9ef8fb102b40f692554c3d0dc493b7abddc97b6665166efcc83bL3038-R3048), [link](https://github.com/fylein/fyle-mobile-app/pull/2145/files?diff=unified&w=0#diff-492aac4c505a9ef8fb102b40f692554c3d0dc493b7abddc97b6665166efcc83bL3080-R3081), [link](https://github.com/fylein/fyle-mobile-app/pull/2145/files?diff=unified&w=0#diff-492aac4c505a9ef8fb102b40f692554c3d0dc493b7abddc97b6665166efcc83bL3115-R3107), [link](https://github.com/fylein/fyle-mobile-app/pull/2145/files?diff=unified&w=0#diff-492aac4c505a9ef8fb102b40f692554c3d0dc493b7abddc97b6665166efcc83bL3986-R3991), [link](https://github.com/fylein/fyle-mobile-app/pull/2145/files?diff=unified&w=0#diff-492aac4c505a9ef8fb102b40f692554c3d0dc493b7abddc97b6665166efcc83bL3996-R4011))
* Remove the unused method `getFormValidationErrors` from `add-edit-expense.page.ts` ([link](https://github.com/fylein/fyle-mobile-app/pull/2145/files?diff=unified&w=0#diff-492aac4c505a9ef8fb102b40f692554c3d0dc493b7abddc97b6665166efcc83bL889-L897))
* Add a new method `showSizeLimitExceededPopover` to `add-edit-expense.page.ts` that shows a popover alert when the receipt size limit is exceeded and change its access modifier to public ([link](https://github.com/fylein/fyle-mobile-app/pull/2145/files?diff=unified&w=0#diff-492aac4c505a9ef8fb102b40f692554c3d0dc493b7abddc97b6665166efcc83bR4291-R4301), [link](https://github.com/fylein/fyle-mobile-app/pull/2145/files?diff=unified&w=0#diff-492aac4c505a9ef8fb102b40f692554c3d0dc493b7abddc97b6665166efcc83bL4304-R4311))
* Add a new file `add-edit-expense.setup.spec.ts` that contains the setup code for the unit tests of the add-edit-expense page and imports the test cases from two other files ([link](https://github.com/fylein/fyle-mobile-app/pull/2145/files?diff=unified&w=0#diff-a763fec878d12e9f0fbac2b5382b030cd01da44167caaf1cff679ba11ffcdfbeR1-R377))

## Clickup
https://app.clickup.com/t/85zt9ph0x

## Code Coverage
```
29.9% Statements 433/1448
15.73% Branches 244/1551
25.35% Functions 108/426
30.45% Lines 426/1399
```

## UI Preview
Please add screenshots for UI changes

### CODE REVIEW

**Review by yksolanki9 — COMMENTED** [review]
Resolve the conflicts in spec file

**Review by yksolanki9 — CHANGES_REQUESTED** [review]
Discussed on call

**Review by Dimple16 — COMMENTED** [review]
PR already reviewed by Yash, will skip,

File: src/app/fyle/add-edit-expense/add-edit-expense-2.spec.ts [hunk 1]
```diff
@@ -0,0 +1,1084 @@
...
+
+    it('reloadCurrentRoute(): should reload the current load', fakeAsync(() => {
+      component.reloadCurrentRoute();
+      tick(500);
+
+      expect(router.navigateByUrl).toHaveBeenCalledOnceWith('/enterprise/my_expenses', { skipLocationChange: true });
+      expect(router.navigate).toHaveBeenCalledOnceWith(['/', 'enterprise', 'add_edit_expense']);
+    }));
+
+    describe('saveExpense():', () => {
```
[the remark this answers is not in the document]
Comment (jayfyle) [reply src/app/fyle/add-edit-expense/add-edit-expense-2.spec.ts:457]: Will add it later, there are some doubts regarding that case

**Pull request closed without merging**